pkg/client/events_tracking.go

```diff
@@ -0,0 +1,92 @@
+package client
+
+import (
+	"encoding/hex"
+	"fmt"
+	"sync"
+
+	"github.com/ethereum/go-ethereum/common"
+)
+
+// requestedSignersTrack is used to track signers generation started after keep
+// creation event is received. It is used to ensure that the process execution
+// is not duplicated, e.g. when the client receives the same event multiple times.
+// When event is received, it should be noted in in this struct. When the signer
+// generation process completes (no matter if it succeeded or failed), it should
+// be removed from this struct.
+type requestedSignersTrack struct {
+	data  map[string]bool // <keep, bool>
+	mutex *sync.Mutex
+}
+
+func (rst *requestedSignersTrack) add(keepAddress common.Address) error {
```

Comment:
Instead of returning an `error`, what do you think about returning a `bool`?

Two advantages I can think of:
- `failed to register signature requested event` does not really bring any value; we'd be able to do just:
    ```
     if !requestedSigners.add(event.KaepAddress) {
       logger.Errorf("signatur generation already requested for keep ...")
     }
    ```

- tests could be simpler as we don't have to check the internal state; so instead of:

    ```
    if !rs.data[keepAddress1.String()] {
		t.Errorf(
			"unexpected value for keep [%s]\nexpected: [%v]\nactual:   [%v]",
			keepAddress1.String(),
			true,
			rs.data[keepAddress1.String()],
		)
	}
    ```

    we just assert `bool` value returned by this function.

Reply:
Sure, good idea. Error is not definitely not needed here.